src/System.Collections.Generic.MultiValueDictionary/src/System/Collections/Generic/MultiValueDictionary.cs

```diff
@@ -0,0 +1,1102 @@
+        ** Variables
+        ======================================================================*/
+
+        /// <summary>
+        /// The private dictionary that this class effectively wraps around
+        /// </summary>
+        private Dictionary<TKey, InnerCollectionView> dictionary;
+
+        /// <summary>
+        /// The function to construct a new <see cref="ICollection{TValue}"/>
+        /// </summary>
+        /// <returns></returns>
+        private Func<ICollection<TValue>> NewCollectionFactory = () => new List<TValue>();
+
+        /// <summary>
+        /// The current version of this MultiValueDictionary used to determine MultiValueDictionary modification
+        /// during enumeration
+        /// </summary>
+        private int version;
+
+        #endregion
+
+        #region Constructors
+        /*======================================================================
+        ** Constructors
+        ======================================================================*/
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MultiValueDictionary{TKey, TValue}" /> 
+        /// class that is empty, has the default initial capacity, and uses the default
+        /// <see cref="IEqualityComparer{TKey}" /> for <typeparamref name="TKey"/>.
+        /// </summary>
+        public MultiValueDictionary()
+        {
+            dictionary = new Dictionary<TKey, InnerCollectionView>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MultiValueDictionary{TKey, TValue}" /> class that is 
+        /// empty, has the specified initial capacity, and uses the default <see cref="IEqualityComparer{TKey}"/>
+        /// for <typeparamref name="TKey"/>.
+        /// </summary>
+        /// <param name="capacity">Initial number of keys that the <see cref="MultiValueDictionary{TKey, TValue}" /> will allocate space for</param>
+        /// <exception cref="ArgumentOutOfRangeException">capacity must be >= 0</exception>
+        public MultiValueDictionary(int capacity)
+        {
+            if (capacity < 0)
+                throw new ArgumentOutOfRangeException("capacity", SR.ArgumentOutOfRange_NeedNonNegNum);
+            dictionary = new Dictionary<TKey, InnerCollectionView>(capacity);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MultiValueDictionary{TKey, TValue}" /> class 
+        /// that is empty, has the default initial capacity, and uses the 
+        /// specified <see cref="IEqualityComparer{TKey}" />.
+        /// </summary>
+        /// <param name="comparer">Specified comparer to use for the <typeparamref name="TKey"/>s</param>
+        /// <remarks>If <paramref name="comparer"/> is set to null, then the default <see cref="IEqualityComparer" /> for <typeparamref name="TKey"/> is used.</remarks>
+        public MultiValueDictionary(IEqualityComparer<TKey> comparer)
```

Comment:
We could, but I'm not sure that we would want to. Wouldn't that limit the functionality of the method since it would constrain the user to making sure their comparer inherited from EqualityComparer when we only really need it to inherit from IEqualityComparer?

I modeled this code after the constructors for [Dictionary<TKey, TValue>](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/Dictionary.cs) so whatever we decide would be similarly relevant there.